nestauk/asf_online_data_exploration
# Data collection pipeline

---

# Description

This PR adds a data collection pipeline for the ASF online data exploration project (for Twitter and The Guardian APIs; I'm still doing further tests to Media Cloud, so that will go out in a different PR).

Fixes #1 

# Instructions for Reviewer

Hey @ch-williamson, thanks so much for taking the time to review the data collection rules. You can see them in `asf_online_data_exploration/config/data_collection_parameters.py`Just wondering if all makes sense in terms of the logic. Note: logical AND is represented by space in Twitter and by AND in The Guardian.

Hey @lizgzil, thanks so much for taking the time to review the data collection pipeline for this project.

In order to test the code in this PR you need to:

- clone this repo: `git clone git@github.com:nestauk/asf_online_data_exploration.git`
- Run `make install`;
- Run `direnv allow`;
- Activate the conda enviroment:`conda activate asf_online_data_exploration`
- Set your API credentials as environment variables
    - *Twitter API credentials:* Run `export BEARER_TOKEN="ADD_YOUR_BEARER_TOKEN_HERE"` in the command line and replace `ADD_YOUR_BEARER_TOKEN_HERE` with your bearer token credentials.
    - *The Guardian credentials*: Run `export GUARDIAN_API_KEY="test"` in the command line.

I would ask you to run the follow scripts I created to test the data collection pipeline scripts:

- `pytest asf_online_data_exploration/pipeline/data_collection/tests/testing_recent_search_twitter.py` → should take approximately 4 seconds to run
- `pytest asf_online_data_exploration/pipeline/data_collection/tests/testing_the_guardian.py` → should take approximately 4 minutes to run

I would also ask you to review the remain scripts (*without running them*), with a bigger focus on the following scripts, to check if the logic behind the data collection pipeline makes sense:

- `asf_online_data_exploration/pipeline/data_collection/recent_search_twitter.py`
- `asf_online_data_exploration/pipeline/data_collection/the_guardian.py`
- `asf_online_data_exploration/utils/data_collection_utils.py`
- `asf_online_data_exploration/config/data_collection_parameters.py` (no need to read the rules in detail, just the general organisation of the file, as Chris will look at the rules in detail).

Any suggestions are welcome - logic, best practices, etc. No need to spend much time looking at the `recent_search_twitter.py` script, as free access to the Twitter APIs expires on the 13th of February.

Again, thanks very much to both!

# Checklist:
